common/hash_data_adapter_hidden_files.go

```diff
@@ -6,15 +6,18 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"sync"
 )
 
 type HiddenFileDataAdapter struct {
 	hashBasePath string // "" == dataBasePath
 	dataBasePath string
+
+	createHashDirOnce *sync.Once
```

Comment:
Should we use this in the lines added below to create the hash dir? I am missing where this is used currently.

Reply:
It was a remnant of prior code changes in this PR-- I forgot to remove this, but, I'll trim it now.

Long story short-- We can't know the entire directory structure up front here, so we have to create it as we go.